src/java.compiler/share/classes/javax/lang/model/util/Elements.java

```diff
@@ -761,4 +761,100 @@ default RecordComponentElement recordComponentFor(ExecutableElement accessor) {
         }
         return null;
     }
+
+
+    /**
+     * {@return the file object for this element or {@code null} if
+     * there is no such file object}
+     *
+     * <p>The returned file object is for the <a
+     * href="../element/package-summary.html#accurate_model">reference
+     * representation</a> of the information used to construct the
+     * element. For example, if during compilation or annotation
+     * processing, a source file for class {@code Foo} is compiled
+     * into a class file, the file object returned for the element
+     * representing {@code Foo} would be for the source file and
+     * <em>not</em> for the class file.
+     *
+     * <p>An implementation may choose to not support the
+     * functionality of this method, in which case {@link
+     * UnsupportedOperationException} is thrown.
+     *
+     * <p>In the context of annotation processing, a non-{@code null}
+     * value is returned if the element was included as part of the
+     * initial inputs or the containing file was created during the
+     * run of the annotation processing tool. Otherwise, a {@code
+     * null} may be returned. In annotation processing, if a
+     * {@linkplain javax.annotation.processing.Filer#createClassFile
+     * class file is created}, that class file can serve as the
+     * reference representation for elements.
+     *
+     * <p>If it has a file object, the file object for a package will
+     * be a {@code package-info} file. A package may exist and not
+     * have any {@code package-info} file even if the package is
+     * (implicitly) created during an annotation processing run from
+     * the creation of source or class files in that package.  An
+     * {@linkplain PackageElement#isUnnamed unnamed package} will have
+     * a null file since it cannot be declared in a compilation unit.
```

Comment:
{@code }` around  null ??

Reply:
I'll fix these before pushing and do a follow-up pass over the API for consistent "null" vs "{@code null}" treatment.